spinn_front_end_common/interface/interface_functions/dsg_region_reloader.py

```diff
@@ -29,96 +28,149 @@
+    regions_base_address = txrx.get_cpu_information_from_core(
+        placement.x, placement.y, placement.p).user[0]
+    start_region = get_region_base_address_offset(regions_base_address, 0)
+    table_size = get_region_base_address_offset(
+        regions_base_address, MAX_MEM_REGIONS) - start_region
+    ptr_table = numpy.frombuffer(txrx.read_memory(
+            placement.x, placement.y, start_region, table_size),
+        dtype=DataSpecificationExecutor.TABLE_TYPE)
+    ptr_table = _fix_ptr_table_sizes(ptr_table)
+
+    # Get the data spec executor pointer table; note that this will not
+    # have the right pointers since not all regions will be regenerated
+    # but the checksum and size will be updated there
+    ds_ptr_table = data_spec_executor.get_pointer_table(0)
+    ds_ptr_table_bytes = ds_ptr_table.view("uint8")
+
+    # Get the offset of the checksum and the checksum + size to write
+    chk_off = DataSpecificationExecutor.TABLE_TYPE.fields['checksum'][1]
+    chk_size = DataSpecificationExecutor.TABLE_TYPE.itemsize - chk_off
+
+    # Write the regions to the machine
+    for i, region in enumerate(data_spec_executor.dsef.mem_regions):
+        if region is not None and not region.unfilled:
+            # Verify that the region hasn't grown bigger than it was
+            new_size = ds_ptr_table[i]["n_words"]
+            old_size = ptr_table[i]["n_words"]
+            if old_size != -1 and new_size > old_size:
+                raise Exception(
+                    f"Region {i} of {vertex} has grown from {old_size} bytes "
+                    f"to {new_size} bytes and will overwrite the next region!")
+            txrx.write_memory(
+                placement.x, placement.y, ptr_table[i]["pointer"],
+                region.region_data[:region.max_write_pointer])
+            # Update the checksum and size of the region
+            reg = get_region_base_address_offset(regions_base_address, i)
+            byte_offset = (reg - start_region) + chk_off
+            addr = reg + chk_off
+            txrx.write_memory(
+                placement.x, placement.y, addr,
+                ds_ptr_table_bytes[byte_offset:], chk_size)
+    vertex.set_reload_required(False)
+    return True
+
+
+def _fix_ptr_table_sizes(ptr_table):
+    """ Go through the pointer table and fix sizes where they have been set to
+        0 to avoid checksum checks
     """
-    __slots__ = ["_txrx", "_data_dir"]
-
-    def __init__(self):
-        self._txrx = FecDataView.get_transceiver()
-        self._data_dir = None
-
-    def _run(self):
-        """
-        """
-
-        # build file paths for reloaded stuff
-        run_dir_path = FecDataView.get_run_dir_path()
-        self._data_dir = generate_unique_folder_name(
-            run_dir_path, "reloaded_data_regions", "")
-        if not os.path.exists(self._data_dir):
-            os.makedirs(self._data_dir)
-
-        report_dir = None
-        if get_config_bool("Reports", "write_text_specs"):
```

Comment:
If you get rid of write_text_specs here you should also do it in https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon/blob/c004357a308bf2f0514aa7580fe148b07ae349e0/spinn_front_end_common/utilities/utility_calls.py#L103

Reply:
These lines turned out to be unused in master, precisely because they are used in get_report_writer instead!

Reply:
ok